substrate/frame/support/src/lib.rs

```diff
@@ -886,7 +886,10 @@ pub mod testing_prelude {
 pub mod pallet_prelude {
 	pub use crate::{
 		defensive, defensive_assert,
-		dispatch::{DispatchClass, DispatchResult, DispatchResultWithPostInfo, Parameter, Pays},
+		dispatch::{
```

Comment:
I now realize that it is actually best if we don't alter re-exports that are made here, and instead bring more and more things into `frame::prelude`. So I would resume by reverting all changes here, add the items added to `frame::prelude` manually as new `pub use frame_support::{this, that}`. 

This is because adding dependencies here would also mean that we will need to cause a breaking change in frame_support, as we are altering the public API. 

Also, in action we are seeing that it in this PR, adding items in this prelude is causing random other pallets to break, because of ambiguous imports.

Reply:
So, the guideline that I have written so far: 

> * When doing so, prefer adding the relevant item to `frame_support::pallet_prelude` and `frame_system::pallet_prelude`.

is not a good advice.

Reply:
edits undone in https://github.com/paritytech/polkadot-sdk/pull/5995/commits/bd17cf205f6ddc934436eb2a3d68d63fb7e86b02